client/blocks/support-article-dialog/index.jsx

```diff
@@ -0,0 +1,113 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { flow, get } from 'lodash';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal Dependencies
+ */
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import SupportArticleHeader from 'blocks/support-article-dialog/header';
+import Placeholders from 'blocks/support-article-dialog/placeholders';
+import EmbedContainer from 'components/embed-container';
+import Emojify from 'components/emojify';
+import QueryReaderPost from 'components/data/query-reader-post';
+import QueryReaderSite from 'components/data/query-reader-site';
+import { getPostByKey } from 'state/reader/posts/selectors';
+import { SUPPORT_BLOG_ID } from 'blocks/inline-help/constants';
+import getInlineSupportArticlePostId from 'state/selectors/get-inline-support-article-post-id';
+import getInlineSupportArticlePostUrl from 'state/selectors/get-inline-support-article-post-url';
+import isInlineSupportArticleVisible from 'state/selectors/is-inline-support-article-visible';
+import { closeSupportArticleDialog } from 'state/inline-support-article/actions';
+
+export class SupportArticleDialog extends Component {
+	static propTypes = {
+		closeSupportArticleDialog: PropTypes.func.isRequired,
+		isVisible: PropTypes.bool,
+		post: PropTypes.object,
+		postId: PropTypes.number,
+		postUrl: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	getDialogButtons() {
+		const { postUrl, translate } = this.props;
+		return [
+			postUrl ? (
+				<Button href={ postUrl } target="_blank" primary>
+					{ translate( 'Visit Article' ) } <Gridicon icon="external" size={ 12 } />
+				</Button>
+			) : (
+				<React.Fragment />
```

Comment:
Why do we have an empty fragment here instead of doing nothing?

Reply:
If I'd have done `postUrl && ( <Button... /> )` we could have ended up with `false` in the array. Which, come to think about it wouldn't be a big issue I don't think. We could always just filter out  falseys as we return the array.

Reply:
Thanks! No worries, I was just curious.